qc_opendrive/main.py

```diff
@@ -71,21 +59,87 @@ def execute_checker(
             "Preconditions are not satisfied. Skip the check.",
         )
 
-        return
+        return False
 
-    # Checker definition setting. If not satisfied then set status as SKIPPED and return
-    if required_definition_setting:
-        schema_version = checker_data.schema_version
 
-        splitted_rule_uid = checker.RULE_UID.split(":")
-        if len(splitted_rule_uid) != 4:
-            raise RuntimeError(f"Invalid rule uid: {checker.RULE_UID}")
+def check_version(checker: types.ModuleType, checker_data: models.CheckerData) -> bool:
+    """
+    Check definition setting and applicable version.
+    If not satisfied then set status as SKIPPED or ERROR and return False
+    """
+    schema_version = checker_data.schema_version
 
-        definition_setting = splitted_rule_uid[2]
-        if (
-            schema_version is None
-            or utils.compare_versions(schema_version, definition_setting) < 0
-        ):
+    splitted_rule_uid = checker.RULE_UID.split(":")
+    if len(splitted_rule_uid) != 4:
+        raise RuntimeError(f"Invalid rule uid: {checker.RULE_UID}")
+
+    definition_setting = splitted_rule_uid[2]
+    definition_setting_expr = f">={definition_setting}"
+    match_definition_setting = version.match(schema_version, definition_setting_expr)
+
+    applicable_version = (
+        checker.APPLICABLE_VERSION if hasattr(checker, "APPLICABLE_VERSION") else None
+    )
+
+    # Check whether applicable version specification is valid
+    if applicable_version is not None and not version.is_valid_version_expression(
+        applicable_version
+    ):
+        checker_data.result.set_checker_status(
+            checker_bundle_name=constants.BUNDLE_NAME,
+            checker_id=checker.CHECKER_ID,
+            status=StatusType.ERROR,
+        )
+
+        checker_data.result.add_checker_summary(
+            constants.BUNDLE_NAME,
+            checker.CHECKER_ID,
+            f"The applicable version {applicable_version} is not valid. Skip the check.",
+        )
+
+        return False
+
+    # Check whether definition setting specification is valid
+    if not version.is_valid_version_expression(definition_setting_expr):
+        checker_data.result.set_checker_status(
+            checker_bundle_name=constants.BUNDLE_NAME,
+            checker_id=checker.CHECKER_ID,
+            status=StatusType.ERROR,
+        )
```

Comment:
For now, `RuleType` accepts `1.7` as a valid definition setting, while our checks only accept `1.7.0`. Therefore, we need this condition here for now and decide later whether to enforce the complete `major.minor.patch` in definition setting in rule UID.

Reply:
You are right, and unfortunately this creates a little more ramifications that previously considered (during the design of the UID). 

Using `RuleType` we already make sure that the version is in format `(\d+)(\.\d+)?(\.\d+)?` (not exactly, just for the sake of the comment), thus we can easily parse with:

```py
r = RuleInfo(*args)
semver.VersionInfo(*(r.definition_settings.split("."))
```

This has a severe limitation: `1.7` is interpreted as `1.7.0`. And this kind of parse, while correct from a semantic version point of view, it is incorrect from a semver **specification** point of view, where the comparison operator has an effect on the intended operation: 

 * `<2` is actually the requirement for `<=1.[max].[max]`
 * `>2` is actually the requirement for `>=3.0.0`

The version in the `uid` is `1.7`. It is referring to `1.7.0` or `1.7.[max]`. There are not enough information in the `uid` itself to know, a convention shall be enforced.

So the discussion really pivotes on:

 * if ASAM enforces semver for all the standards than we can enforce the full semantic versioning in rule uid
 * otherwise we can forcibly set the convention that elision means minimum, thus `1.7` in uid **is** `1.7.0`

I want to raise the question to @pmai or @AsamDiegoSanchez.

Reply:
This is interesting and I think we could create an issue from this discussion, @MatteoRagni.

For now, all the [rule UIDs for OpenDrive](https://github.com/asam-ev/qc-opendrive/blob/main/checker_bundle_doc.md) are in the complete form `major.minor.patch`. 

If there is a **rule UID** defined with only `major.minor`, e.g. `1.7`, the status of the check is explicitly set to `ERROR` with the message `The applicable version {applicable_version} is not valid`, which I think is a good temporary compromise so that this issue does not block us from releasing.

If the **input file** only defines `major.minor`, we explicitly append the patch `.0`, so `1.7` becomes `1.7.0`.

Reply:
The discussion is **resolved** for the current PR, I'm not marking it as resolved to not collapse it.

Reply:
Moved [here](https://github.com/asam-ev/qc-framework/issues/203)